packages/modelviewer.dev/examples/scenegraph/index.html

```diff
@@ -400,24 +400,85 @@ <h2 class="demo-title">Swap textures - Normals/Occlusion/Emission</h2>
 
   let material = modelViewerTexture.model.materials[0];
 
-  let applyTexture = (channel, event) => {
-    material[channel].texture.source.setURI(event.target.value);
+  let createAndApplyTexture = async (channel, event) => {
+    let texture = await modelViewerTexture.createTexture(event.target.value);
+    material[channel].setTexture(texture);
   }
 
   document.querySelector('#normals').addEventListener('input', (event) => {
-    applyTexture('normalTexture', event);
+    createAndApplyTexture('normalTexture', event);
   });
   
   document.querySelector('#occlusion').addEventListener('input', (event) => {
-    applyTexture('occlusionTexture', event);
+    createAndApplyTexture('occlusionTexture', event);
   });
   
   document.querySelector('#emission').addEventListener('input', (event) => {
-    applyTexture('emissiveTexture', event);
+    createAndApplyTexture('emissiveTexture', event);
+  });
+});
+
+</script>
+            </template>
+          </example-snippet>
+        </div>
+      </div>
+    </div>
+
+
+
+    <div class="sample">
+      <div id="createTexturesExample" class="demo"></div>
+      <div class="content">
+        <div class="wrapper">
+          <div class="heading">
+            <h2 class="demo-title">Swap textures - Normals/Occlusion/Emission</h2>
+            <p>As above, you can change these values in AR, but only in WebXR
+              mode. iOS Quick Look reflects these texture changes so long as the
+              USDZ is auto-generated.</p>
+          </div>
+          <example-snippet stamp-to="createTexturesExample" highlight-as="html">
+            <template>
+<model-viewer id="duck" camera-controls src="../../shared-assets/models/glTF-Sample-Models/2.0/Duck/glTF-Binary/Duck.glb" ar ar-modes="webxr scene-viewer quick-look" alt="A 3D model of a helmet">
+  <div class="controls">
+    <div>
+      <p>Normals</p>
+      <select id="normals2">
+        <option>None</option>
+        <option value="../../shared-assets/models/glTF-Sample-Models/2.0/DamagedHelmet/glTF/Default_normal.jpg">Damaged helmet</option>
+        <option value="../../shared-assets/models/glTF-Sample-Models/2.0/Lantern/glTF/Lantern_normal.png">Lantern Pole</option>
+        <option value="../../shared-assets/models/glTF-Sample-Models/2.0/WaterBottle/glTF/WaterBottle_normal.png">Water Bottle</option>
+      </select>
+    </div>
+</model-viewer>
+<script type="module">
+const modelViewerTexture = document.querySelector("model-viewer#duck");
+
+modelViewerTexture.addEventListener("load", () => {
+
+  let material = modelViewerTexture.model.materials[0];
+
+  let createAndApplyTexture = async (channel, event) => {
+    if(event.target.value == "None"){
+      // Clears the texture.
```

Comment:
Nice, I look forward to trying this out!

Reply:
:)